src/content/docs/en/reference/integrations-reference.mdx

```diff
@@ -453,164 +510,137 @@ It allows you to have a dedicated folder, avoiding conflicts with another integr
-                const projectRoutes = routes.filter(r => r.origin === 'project').map(r => r.pattern)
-                
-                console.log(projectRoutes)
-            },
+##### `route.component`
+
+<p>
+**Type:** `string`<br />
+<Since v="4.14.0" />
+</p>
+
+The `component` property is a readonly string path relative to the project root, e.g `"src/pages/blog/[slug].astro"`. Its value helps you identify the route being set up.
+
+##### `route.prerender`
+
+<p>
+**Type:** `boolean`<br />
+**Default:** `undefined`<br />
+<Since v="4.14.0" />
+</p>
+
+The `prerender` property is used to configure [on-demand server rendering](/en/guides/on-demand-rendering/#enabling-on-demand-rendering) for a route. If the route file contains an explicit `export const prerender` value, the value will be used as the default instead of `undefined`.
+
+```js title="astro.config.mjs"
+import { defineConfig } from 'astro/config';
+
+export default defineConfig({
+  integrations: [setPrerender()],
+});
+
+function setPrerender() {
+  return {
+    name: 'set-prerender',
+    hooks: {
+      'astro:route:setup': ({ route }) => {
+        if (route.component.endsWith('/blog/[slug].astro')) {
+          route.prerender = true;
         }
-    }
+      },
+    },
+  };
 }
 ```
+ 
+If the final value after running all the hooks is `undefined`, the route will fall back to a prerender default based on the [`output` option](/en/reference/configuration-reference/#output): prerendered for `static` mode, and on-demand rendered for `server` mode.
 
-##### `IntegrationResolvedRoute` type reference
+### `astro:routes:resolved`
 
-```ts
-interface IntegrationResolvedRoute {
-	/**
-	 * The current **pattern** of the route. For example:
-	 * - `src/pages/index.astro` has a pattern of `/`
-	 * - `src/pages/blog/[...slug].astro` has a pattern of `/blog/[...slug]`
-	 * - `src/pages/site/[blog]/[...slug].astro` has a pattern of `/site/[blog]/[...slug]`
-	 */
-	pattern: RouteData['route'];
+<p>
 
-	/**
-	 *
-	 * regex used for matching an input URL against a requested route
-	 * ex. "[fruit]/about.astro" will generate the pattern: /^\/([^/]+?)\/about\/?$/
-	 * where pattern.test("banana/about") is "true"
-	 *
-	 * ## Example
-	 *
-	 * ```js
-	 * if (route.pattern.test('/blog')) {
-	 *  // do something
-	 * }
-	 * ```
-	 */
-	patternRegex: RouteData['pattern'];
+<Since v="5.0.0" />
+</p>
 
-	/**
-	 *  Source component URL
-	 */
-	entrypoint: RouteData['component'];
+**Previous hook:** [`astro:route:setup`](#astroroutesetup)
 
-	/**
-	 * Whether the route is prerendered or not
-	 */
-	isPrerendered: RouteData['prerender'];
+**Next hook:** [`astro:config:done`](#astroconfigdone)
 
-	/**
-	 * The {@link IntegrationResolvedRoute} to redirect to. It's present when `IntegrationResolvedRoute.type` is `redirect`.
-	 */
-	redirectRoute?: IntegrationResolvedRoute;
+**When:** In `astro dev`, it also runs if a file based route changes (added/removed/updated).
```

Comment:
```suggestion
**When:** In `astro dev`, it also runs on every change to a file based route (added/removed/updated).
```

I think it is better to note that it can run multiple times, not once in case an update happens.